crates/bevy_asset/src/io/embedded/mod.rs

```diff
@@ -30,7 +30,13 @@ impl EmbeddedAssetRegistry {
     /// running in a non-rust file). `asset_path` is the path that will be used to identify the asset in the `embedded`
     /// [`AssetSource`]. `value` is the bytes that will be returned for the asset. This can be _either_ a `&'static [u8]`
     /// or a [`Vec<u8>`].
-    #[allow(unused)]
+    #[cfg_attr(
+        not(feature = "embedded_watcher"),
+        expect(
+            unused_variables,
+            reason = "The `full_path` argument is not used when `embedded_watcher` is disabled."
+        )
+    )]
```

Comment:
Can we improve the API by instead changing the function signature on this feature?

Reply:
It seems likely, but should we create an issue rather than folding an API change into this PR?

Reply:
Yeah, would you like to do that for each of those?

Reply:
No, I think I have my hands full already :sweat_smile:

Reply:
Ok, I can create them after this is merged. I'll wait because there might be more and it'll be easier to do all at once.